waggle-dance-api/src/main/java/com/hotels/bdp/waggledance/api/model/PrimaryMetaStore.java

```diff
@@ -47,6 +50,24 @@ public FederationType getFederationType() {
     return FederationType.PRIMARY;
   }
 
+  @Override
+  public List<String> getMappedDatabases() {
+    return mappedDatabases;
+  }
+
+  @Override
+  public void setMappedDatabases(List<String> mappedDatabases) {
+    if (mappedDatabases != null) {
```

Comment:
Not sure if this is needed,

Reply:
If I don't have it, I get a nullpointer exception instead of a validation error for `mapped-databases: `

Reply:
Because it does `null.isEmpty()`

Reply:
Why don't we treat `mapped-databases:` as null
This weird stuff with the if something is not set we treat it as null via this boolean but we actually return an emptyList is just very confusing. :)

Reply:
With not needed I mean the whole if block :). Just set the value and leave all logic out of this class. I feel it belongs in the Whitelist class that should deal with null, empty etc...